src/PKSim.Core/Snapshots/Mappers/ProjectMapper.cs

```diff
@@ -222,16 +220,18 @@ private void addQualificationPlanToProject(ModelProject project, Model.Qualifica
 
       private Task<Model.Simulation[]> allSmulationsFrom(Simulation[] snapshotSimulations, ModelProject project) => _simulationMapper.MapToModels(snapshotSimulations, project);
 
-      private Task<IPKSimBuildingBlock[]> allBuidingBlocksFrom(SnapshotProject snapshot)
+      private async Task<IEnumerable<IPKSimBuildingBlock>> allBuidingBlocksFrom(SnapshotProject snapshot)
       {
-         var tasks = new List<Task<object>>();
```

Comment:
This was running all those tasks in //. Great until I realized that there is one instance of the mapper used per building block, which is good. But until you realized that loading many populations at the same time kinds of destroy other populations. That way, each building block is loaded one at at time. No issue with concurrent load